compiler/utils.go

```diff
@@ -643,3 +644,38 @@ func endsWithReturn(stmts []ast.Stmt) bool {
 func encodeIdent(name string) string {
 	return strings.Replace(url.QueryEscape(name), "%", "$", -1)
 }
+
+// formatJSStructTagVal returns a string of Javascript code appropriate for
+// accessing the property identified by jsTag. If the jsTag value can be safely
+// encoded in Javascript using the dot notation this is used, else we fallback
+// to the bracket notation. For more details see:
+//
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors
+//
+// Uses definition of an identifier from
+// https://developer.mozilla.org/en-US/docs/Glossary/Identifier
+func formatJSStructTagVal(jsTag string) string {
+	useDot := true
+
+Outer:
+	for i, r := range jsTag {
+		switch i {
+		case 0:
+			if r != '$' && r != '_' && !unicode.IsLetter(r) {
+				useDot = false
+				break Outer
+			}
+		default:
+			if r != '$' && r != '_' && !unicode.IsLetter(r) && !unicode.IsNumber(r) {
+				useDot = false
+				break Outer
+			}
+		}
+	}
+
+	if useDot {
+		return "." + jsTag
+	}
+
+	return "[\"" + jsTag + "\"]"
```

Comment:
Don't we need escaping some chars like backslash? `text/template.JSEscapeString` is needed here.

Reply:
Hmm, there are very subtle differences between `strconv.Quote` and `template.JSEscapeString`...

1. `<` and `>` is treated as special chars in `JSEscapeString`
2. `strconv.Quote` can produce \U000XXXXX for non-BMP and non-printable char, which is not valid escaping in ES5

Reply:
You're absolutely right - thank you for correcting me!

I've just added a test in the latest commit that does a `.Get` in order to ensure the property that has been set via the js struct tag is using the expected key.

PTAL